File: app/src/main/java/com/demosample/demorizzle/repository/DataRepository.java
package com.demosample.demorizzle.repository;

import android.app.Application;

import androidx.lifecycle.MutableLiveData;

import com.demosample.demorizzle.model.BaseResponse;
import com.google.gson.Gson;

import java.io.IOException;
import java.io.InputStream;

public class DataRepository {

    private static DataRepository myApiRepo;
    private static Application context;


    public DataRepository() {
    }

    public static DataRepository getInstance(Application application) {
        if (myApiRepo == null) {
            {
                myApiRepo = new DataRepository();
                context = application;
            }

        }
        return myApiRepo;
    }

    public String readJSONFromAsset() {
        String json = null;
        try {
            InputStream is = context.getAssets().open("abc.json");
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            json = new String(buffer, "UTF-8");
        } catch (IOException ex) {
            ex.printStackTrace();
            return null;
        }
        return json;
    }

    public MutableLiveData<BaseResponse> getDetails() {
        final MutableLiveData<BaseResponse> listData = new MutableLiveData<>();
        Gson gson = new Gson();
        //    JSONObject obj = new JSONObject(readJSONFromAsset());
        BaseResponse baseResponse = gson.fromJson(readJSONFromAsset(), BaseResponse.class);
        listData.setValue(baseResponse);
        return listData;
    }

}

File: app/src/main/java/com/demosample/demorizzle/repository/DataRepository.kt
package com.demosample.demorizzle.repository

import android.app.Application
import androidx.lifecycle.MutableLiveData
import com.demosample.demorizzle.model.BaseResponse
import com.google.gson.Gson
import java.io.IOException
import java.nio.charset.Charset

class DataRepository {
    private fun readJSONFromAsset(): String? {
        var json: String? = null
        json = try {
            val `is` = context!!.assets.open("abc.json")
            val size = `is`.available()
            val buffer = ByteArray(size)
            `is`.read(buffer)
            `is`.close()
            String(buffer, Charset.forName("UTF-8"))
        } catch (ex: IOException) {
            ex.printStackTrace()
            return null
        }
        return json
    }

    //    JSONObject obj = new JSONObject(readJSONFromAsset());
    val details: MutableLiveData<BaseResponse>
        get() {
            val listData = MutableLiveData<BaseResponse>()
            val gson = Gson()
            //    JSONObject obj = new JSONObject(readJSONFromAsset());
            val baseResponse = gson.fromJson(readJSONFromAsset(), BaseResponse::class.java)
            listData.value = baseResponse
            return listData
        }

    companion object {
        private var myApiRepo: DataRepository? = null
        private var context: Application? = null
        fun getInstance(application: Application?): DataRepository? {
            if (myApiRepo == null) {
                run {
                    myApiRepo = DataRepository()
                    context = application
                }
            }
            return myApiRepo
        }
    }
}